### PR TITLE
Delay an `is_local_ever_initialized` call.

### PR DESCRIPTION
**Pull request by nnethercote**
This commit moves the call after a `return` that almost always runs. It
speeds up the `unicode_normalization` benchmark by about 2%.

r? @spastorino

### CODE REVIEW

File: src/librustc_mir/borrow_check/mod.rs [hunk 1]
```diff
@@ -1966,8 +1956,18 @@ impl<'cx, 'tcx> MirBorrowckCtxt<'cx, 'tcx> {
             }
         }
 
+        // rust-lang/rust#21232, #54986: during period where we reject
```
Comment (Centril) [inline src/librustc_mir/borrow_check/mod.rs]: The "during period" comment is outdated now that the period is over? cc @tmandry @pnkfelix

**Comment by nnethercote** [comment]
The CI perf results match the local ones: `unicode_normalization` instructions counts are reduced by up to 2.4%. Other benchmarks are unaffected.

**Review by spastorino — APPROVED** [review]

**Comment by spastorino** [comment]
Just talked to @Centril about this PR and can continue with his comments in a follow up PR or issue.
@pnkfelix || @tmandry can you check that?.

@bors r+

File: src/librustc_mir/borrow_check/mod.rs (hunk 1 above)
Comment (pnkfelix) [reply src/librustc_mir/borrow_check/mod.rs]: What? #54987 is still open, and as far as I know we are still rejecting attempts to partially initialize... is that not the case? I just now noticed the additional points added by Ralf and others about how that feature would interact with ideas/plans to make things like `(String, !)` a ZST, too... so the comment seems still relevant to me. But I haven’t tried anything in the playpen yet...

Comment (Centril) [reply src/librustc_mir/borrow_check/mod.rs]: Ah; on re-reading the comment it seems its not about the warning we emitted back then but which we turned into a hard error. Makes sense now.

**Pull request merged**